File: cslacs/Csla/Serialization/Mobile/NullPlaceholder.cs
using System;

namespace Csla.Serialization.Mobile
{
  /// <summary>
  /// Placeholder for null child objects.
  /// </summary>
  [Serializable()]
  public sealed class NullPlaceholder : IMobileObject
  {
    #region Constructors

    public NullPlaceholder()
    {
      // Nothing
    }

    #endregion

    #region IMobileObject Members

    public void GetState(SerializationInfo info)
    {
      // Nothing
    }

    public void GetChildren(SerializationInfo info, MobileFormatter formatter)
    {
      // Nothing
    }

    public void SetState(SerializationInfo info)
    {
      // Nothing
    }

    public void SetChildren(SerializationInfo info, MobileFormatter formatter)
    {
      // Nothing
    }

    #endregion
  }
}

File: cslavb/Csla/Serialization/Mobile/NullPlaceholder.vb

Namespace Serialization.Mobile

  ''' <summary>
  ''' Placeholder for null child objects.
  ''' </summary>
  ''' <remarks></remarks>
  <Serializable()> _
  Public NotInheritable Class NullPlaceholder
    Implements IMobileObject


#Region " Constructors "

    Public Sub New()
      'Nothing
    End Sub

#End Region

#Region " IMobileObject Members "

    Public Sub GetChildren(ByVal info As SerializationInfo, ByVal formatter As MobileFormatter) Implements IMobileObject.GetChildren
      'Nothing
    End Sub

    Public Sub GetState(ByVal info As SerializationInfo) Implements IMobileObject.GetState
      'Nothing
    End Sub

    Public Sub SetChildren(ByVal info As SerializationInfo, ByVal formatter As MobileFormatter) Implements IMobileObject.SetChildren
      'Nothing
    End Sub

    Public Sub SetState(ByVal info As SerializationInfo) Implements IMobileObject.SetState
      'Nothing
    End Sub

#End Region
    
  End Class

End Namespace

